payment/order.go

```diff
@@ -39,71 +58,56 @@ type OrderItem struct {
-			// Staging - User Wallet Vote
-			"AgEJYnJhdmUuY29tAiNicmF2ZSB1c2VyLXdhbGxldC12b3RlIHNrdSB0b2tlbiB2MQACFHNrdT11c2VyLXdhbGxldC12b3RlAAIKcHJpY2U9MC4yNQACDGN1cnJlbmN5PUJBVAACDGRlc2NyaXB0aW9uPQACGmNyZWRlbnRpYWxfdHlwZT1zaW5nbGUtdXNlAAAGIOH4Li+rduCtFOfV8Lfa2o8h4SQjN5CuIwxmeQFjOk4W",
-			// Staging - Anon Card Vote
-			"AgEJYnJhdmUuY29tAiFicmF2ZSBhbm9uLWNhcmQtdm90ZSBza3UgdG9rZW4gdjEAAhJza3U9YW5vbi1jYXJkLXZvdGUAAgpwcmljZT0wLjI1AAIMY3VycmVuY3k9QkFUAAIMZGVzY3JpcHRpb249AAIaY3JlZGVudGlhbF90eXBlPXNpbmdsZS11c2UAAAYgPV/WYY5pXhodMPvsilnrLzNH6MA8nFXwyg0qSWX477M=",
-			// Staging - Free Trial
-			"MDAxN2xvY2F0aW9uIGJyYXZlLmNvbQowMDJkaWRlbnRpZmllciBicmF2ZSBmcmVlLXRyaWFsIHNrdSB0b2tlbiB2MQowMDE3Y2lkIHNrdT1mcmVlLXRyaWFsCjAwMTBjaWQgcHJpY2U9MAowMDE1Y2lkIGN1cnJlbmN5PUJBVAowMDM0Y2lkIGRlc2NyaXB0aW9uPUdyYW50cyByZWNpcGllbnQgb25lIGZyZWUgdHJpYWwKMDAyM2NpZCBjcmVkZW50aWFsX3R5cGU9c2luZ2xlLXVzZQowMDJmc2lnbmF0dXJlIGfeOulgTyOWVP1Qiszt8lfPnppPJQhoi8xTfI6bzqO4Cg==":
-			return true
-		}
+func decodeAndUnmarshalSku(sku string) (*macaroon.Macaroon, error) {
+	macBytes, err := macaroon.Base64Decode([]byte(sku))
+	if err != nil {
+		return nil, fmt.Errorf("failed to b64 decode sku token: %w", err)
 	}
-
-	whitelistedSKUs := strings.Split(os.Getenv("SKUS_WHITELIST"), ",")
-	for _, whitelistedSKU := range whitelistedSKUs {
-		if sku == whitelistedSKU {
-			return true
-		}
+	mac := &macaroon.Macaroon{}
+	if err = mac.UnmarshalBinary(macBytes); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal sku token: %w", err)
 	}
 
-	return false
+	return mac, nil
 }
 
 // CreateOrderItemFromMacaroon creates an order item from a macaroon
-func CreateOrderItemFromMacaroon(sku string, quantity int) (*OrderItem, error) {
-	macBytes, err := macaroon.Base64Decode([]byte(sku))
+func (s *Service) CreateOrderItemFromMacaroon(sku string, quantity int) (*OrderItem, *Methods, error) {
+	mac, err := decodeAndUnmarshalSku(sku)
 	if err != nil {
-		return nil, err
+		return nil, nil, fmt.Errorf("failed to create order item from macaroon: %w", err)
 	}
-	mac := &macaroon.Macaroon{}
-	err = mac.UnmarshalBinary(macBytes)
+
+	// get the merchant's keys
+	keys, err := s.Datastore.GetKeys(mac.Location(), false)
 	if err != nil {
-		return nil, err
+		return nil, nil, fmt.Errorf("failed to get keys for merchant to validate macaroon: %w", err)
+	}
+
+	// check if any of the keys for the merchant will validate the mac
+	var valid bool
+	for _, k := range *keys {
+		// decrypt the merchant's secret key from db
+		if err := k.SetSecretKey(); err != nil {
+			return nil, nil, fmt.Errorf("unable to decrypt merchant key from db: %w", err)
+		}
+		// perform verify
+		if _, err := mac.VerifySignature([]byte(k.SecretKey), nil); err == nil {
+			// valid
+			valid = true
+		}
```

Comment:
I think there is a bug here. Possible fix:
```suggestion
		if _, err := mac.VerifySignature([]byte(k.SecretKey), nil); err == nil {
			// valid
			valid = true
		} else {
		        return nil, nil, ErrInvalidSKU
		}
```

Lines 104 through 107 should be deleted as a result

Reply:
Ah, wouldn't we want to break it out of the loop then?

Reply:
sure.  resolved: https://github.com/brave-intl/bat-go/pull/816/commits/1339f8c4034e560a9231615dab508dfc0cd0233f